client/widgets/Images.cpp

```diff
@@ -259,10 +290,26 @@ void CAnimImage::showAll(SDL_Surface * to)
 
 	if(flags & CShowableAnim::BASE && frame != 0)
 		if(auto img = anim->getImage(0, group))
-			img->draw(to, pos.x, pos.y);
+		{
+			if (isScaled())
+			{
+				auto scaled = img->scaleFast(float(scaledSize.x) / img->width());
```

Comment:
Scaled image should be cached

Reply:
Maybe I'll just add new resource file with the new icon size instead. This dialog needs a bunch of new icons anyway.